### PR TITLE
Add contribution guide and a code of conduct

### PR DESCRIPTION
**Pull request by nbsp**
@davidzhao add conduct@ email alias, and merge

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+We as members, contributors, and leaders pledge to make participation in our
+community a harassment-free experience for everyone, regardless of age, body
+size, visible or invisible disability, ethnicity, sex characteristics, gender
+identity and expression, level of experience, education, socio-economic status,
+nationality, personal appearance, race, caste, color, religion, or sexual
+identity and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
```
Comment (davidzhao) [inline CODE_OF_CONDUCT.md]:
```suggestion
We are committed to providing a welcoming, respectful, and harassment-free environment for everyone, regardless of background, experience, or identity. We strive to foster a positive and inclusive community where all participants feel valued and empowered to contribute.
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
```
Comment (davidzhao) [inline CODE_OF_CONDUCT.md]:
```suggestion
### Expected behavior
```

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or advances of
+  any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email address,
+  without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
```
Comment (davidzhao) [inline CODE_OF_CONDUCT.md]:
```suggestion
### Unacceptable Behavior

* Harassment, discrimination, or offensive comments regarding identity, appearance, or background.
* Publishing others' private information, such as a physical or email address,
  without their explicit permission
* Personal attacks, insults, or disruptive behavior that undermines the community.
* Posting content or engaging in activities that are inappropriate, unlawful, or harmful.
```

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+
+**Community Impact**: Demonstrating a pattern of violation of community
+standards, including sustained inappropriate behavior, harassment of an
+individual, or aggression toward or disparagement of classes of individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction within the
+community.
```
Comment (davidzhao) [inline CODE_OF_CONDUCT.md]: IMO less is more here

```suggestion
Violations of this Code of Conduct may result in removal from the community, project, or repository. Severe violations may result in a permanent ban.
```

**Review by davidzhao — COMMENTED** [review]
a few comments on CoC. otherwise lg.

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (nbsp) [reply CODE_OF_CONDUCT.md]: i think it's better to not change this CoC, as the Contributor Covenant is more or less the industry standard across thousands of big projects, and i'm worried that changing it subtly can lead to confusions with people who think it's the covenant and don't read too much into it

Comment (nbsp) [reply CODE_OF_CONDUCT.md]: we can always have a note on top saying this is based on the covenant but with some changes though — your call

Comment (davidzhao) [reply CODE_OF_CONDUCT.md]: I would prefer to start concise and with minimal provisions. we are not a large project (yet) and IMO it doesn't make sense to adopt processes of such large projects.

Comment (nbsp) [reply CODE_OF_CONDUCT.md]: makes sense. added the following line to the attribution:

```
It has been subtly adapted for formatting and brevity, as well as changing the
actions taken after a violation.
```

**Review by davidzhao — APPROVED** [review]

**Comment by davidzhao** [comment]
lgtm! I'll get the alias set up later today

**Pull request merged**